wordpress-mobile/WordPress-Android
# Site Creation: Domains Screen

Fixes #8507. This PR is a full re-write of the domains screen. It follows the patterns established by the verticals screen and behaves almost exactly the same way. It also removes the unused domains components from the new site creation flow. The unit tests are not included and will hopefully be tackled in a separate PR.

@SylvesterWilmott We'll be tackling the tablet layout in a separate issue #8735. Here are portrait and landscape screenshots for phones: https://cloudup.com/c6IXDSVVH-l

**To get to new domains page:**
* Ensure you're using the debug build so the new site creation flow is used
* Go to site selection screen
* Tap on `+` menu button
* Select `Create WordPress.com site`
* Select any segment
* Select a vertical or tap `Skip`
* Fill in the site info and tap `Next` or simply `Skip`

**During test:**
Remember to rotate your screen from time to time

**To test:**
* Type anything and verify domain suggestions show up and the header is hidden
* Select a domain and verify it's selected and `Create site` button shows up
* Select another domain and verify other domain is de-selected
* Type anything else and verify no domain is selected after reload (and the `create site` button is hidden)
* Tap on the `X` button on the right of the edit text (while there is some text) and verify the current text is deleted and the header is shown
* Disable your network, type anything and verify that the error row is loaded stating there is no network
* Enable your network and tap on the refresh button and verify that domain suggestions are loaded
* Tap on the help button and verify that the help screen is shown
* Go back and verify that site info screen is shown
* Tap on the `Create Site` button for any domain suggestion (don't worry it's not yet connected) and verify that the site preview screen is shown

P.S: I am happy to separate this PR into 2 (or more), one for the UI and the other for the view model. However, I think it'll be easy enough to review this since it's following an established pattern and a lot of the changes are for layout anyway.
